src/app/public/testing/phonefield/phone-field-fixture.ts

```diff
@@ -0,0 +1,276 @@
+    const inputEl = this.phoneFieldInput;
+    inputEl.value = inputText;
+
+    SkyAppTestUtility.fireDomEvent(inputEl, 'input');
+    this.fixture.detectChanges();
+
+    SkyAppTestUtility.fireDomEvent(inputEl, 'change');
+
+    this.fixture.detectChanges();
+    return this.fixture.whenStable();
+  }
+
+  /**
+   * Opens the country selector, peforms a search, but makes no selection.
+   * @param searchText The name of the country to select.
+   */
+  public async searchCountry(searchText: string): Promise<NodeListOf<HTMLElement>> {
+    await this.openCountrySelection();
+
+    const countryFixture = await this.getCountryFixture();
+    const results = await countryFixture.search(searchText);
+
+    await this.waitForCountrySelection();
+    return results;
+  }
+
+  /**
+   * Opens the country selector, performs a search, and selects the first result (if any).
+   * @param searchText The name of the country to select.
+   */
+  public async selectCountry(searchText: string): Promise<any> {
+    await this.openCountrySelection();
+
+    const countryFixture = await this.getCountryFixture();
+    await countryFixture.select(searchText);
+
+    return this.waitForCountrySelection();
+  }
+
+  //#region helpers
+
+  private get countryElement(): HTMLInputElement {
+    return this._debugEl.query(By.css('sky-country-field')).nativeElement;
+  }
+
+  private get countryFlagButton(): HTMLInputElement {
+    return this._debugEl.query(By.css('.sky-phone-field-country-btn .sky-btn')).nativeElement;
+  }
+
+  private get phoneFieldInput(): HTMLInputElement {
+    return this._debugEl.query(By.css('input[skyPhoneFieldInput]')).nativeElement;
+  }
+
+  private async getCountryFixture(): Promise<SkyCountryFieldFixture> {
+    if (this._countryFixture === undefined) {
+      // tag the country field with a sky test id
+      const countrySkyTestId = `${this.skyTestId}-country`;
+      this.setSkyTestId(this.countryElement, countrySkyTestId);
+
+      // initialize the country fixture
+      this._countryFixture = new SkyCountryFieldFixture(this.fixture, countrySkyTestId);
+
+      this.fixture.detectChanges();
+      await this.fixture.whenStable();
+    }
+
+    return this._countryFixture;
+  }
+
+  private async openCountrySelection(): Promise<any> {
+    this.countryFlagButton.click();
+    return this.waitForCountrySelection();
+  }
+
+  private setSkyTestId(element: HTMLElement, skyTestId: string) {
+    element.setAttribute('data-sky-id', skyTestId);
+  }
+
+  private async waitForCountrySelection(): Promise<any> {
+    // any country selection needs extra time to complete
+    this.fixture.detectChanges();
+    await this.fixture.whenStable();
+
+    this.fixture.detectChanges();
+    return this.fixture.whenStable();
+  }
+
+  //#endregion
+}
+
+// TODO: REMOVE ME ONCE WE PUBLISH THE COUNTRY FIELD!!!!!
+
+/**
+ * Allows interaction with a SKY UX country field component.
+ */
+export class SkyCountryFieldFixture {
```

Comment:
This is copy/pasted for now while I'm waiting for the country field fixture to be published so I can update my package.json.